### ISSUE TITLE
[P1] Make the PPO/config contract executable: declared settings are ignored and typoed overrides succeed

### ISSUE THREAD

**Issue by ankaggarwal94** [ankaggarwal94/qanta-buzzer#36]
## Summary

The configuration surface currently overstates what the training code actually controls.

At current `main` (`5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`):

- `configs/default.yaml` declares PPO values including:
  - `gae_lambda: 0.95`
  - `clip_ratio: 0.2`
  - `value_coef: 0.5`
  - `entropy_coef: 0.01`
  - `max_grad_norm: 0.5`
  - `target_kl: 0.03`
- `scripts/train_ppo.py` passes only learning rate, `n_steps`, batch size, epochs, gamma, seed, and network architecture to `PPOBuzzer`.
- `PPOBuzzer.__init__()` does not accept or forward the omitted values to Stable-Baselines3.
- As a result, the actual SB3 model uses library defaults. Materially, configured `entropy_coef: 0.01` becomes SB3's default `ent_coef = 0.0`, and configured `target_kl: 0.03` becomes `None`.
- `qb_data.config.merge_overrides()` creates unknown nested keys instead of rejecting them, so a typo such as `environment.wait_penatly=0.1` is printed as an applied override and persisted even though the environment continues using `wait_penalty`.

This makes `config_used.json` an unreliable statement of the executed experiment. Humans have once again invented configuration files that function partly as wishes.

## Relevant code

- [`configs/default.yaml`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/configs/default.yaml)
- [`scripts/train_ppo.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/scripts/train_ppo.py)
- [`agents/ppo_buzzer.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/agents/ppo_buzzer.py)
- [`qb_data/config.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/qb_data/config.py)

Other declared-but-not-consistently-honored fields should be audited in the same repair, including `likelihood.batch_size`, `likelihood.max_length`, `environment.max_steps`, and `evaluation.run_*` flags.

### Declared PPO coefficient is ignored

```bash
python scripts/train_ppo.py \
  --config configs/smoke.yaml \
  ppo.entropy_coef=123.0 \
  ppo.target_kl=0.000001
```

The resolved config records the values, but `PPOBuzzer` constructs SB3 PPO without `ent_coef` or `target_kl`; the model still uses SB3 defaults.

A direct regression can assert:

```python
assert agent.model.ent_coef == 123.0
assert agent.model.target_kl == 0.000001
```

Those assertions fail on current main.

### Typoed override is accepted

```bash
python scripts/train_ppo.py \
  --config configs/smoke.yaml \
  environment.wait_penatly=999
```

The run accepts and persists a new `wait_penatly` key, while `make_env_from_config()` continues using the original `wait_penalty`.

## Impact

- The default MLP PPO run has no entropy bonus despite the repository declaring `0.01`.
- Hyperparameter sweeps can report distinct configurations that instantiate identical agents.
- `target_kl` safety/early-stopping expectations are not enforced.
- Typoed keys silently yield the old behavior while provenance claims the typoed setting was used.
- Reproduction and comparison artifacts can be syntactically valid but semantically false.

## Expected behavior

Configuration should be a checked executable contract:

1. Every supported key is mapped to the actual runtime API (`clip_ratio -> clip_range`, `value_coef -> vf_coef`, `entropy_coef -> ent_coef`, etc.).
2. Unsupported/dead keys fail validation rather than being persisted as if active.
3. Unknown override paths and type mismatches fail before any expensive work begins.
4. Artifacts record both the requested config and the effective runtime parameters read back from the instantiated model/environment.

## Acceptance criteria

- [ ] Forward all supported PPO fields to SB3/MaskablePPO, including GAE lambda, clip range, value coefficient, entropy coefficient, max grad norm, and target KL.
- [ ] Add a schema or typed configuration layer covering every active section and reject unknown keys by default.
- [ ] Reject misspelled override paths such as `wait_penatly`; provide nearest-key suggestions where practical.
- [ ] Reject wrong value types and invalid ranges before model/data initialization.
- [ ] Audit all fields in `configs/default.yaml`, `configs/smoke.yaml`, and `configs/t5_policy.yaml`; each must be consumed, explicitly deprecated, or removed.
- [ ] Persist an `effective_runtime_config` block derived from the instantiated SB3 model/environment, not merely the YAML object.
- [ ] Add tests using extreme non-default values so accidental reliance on coincidentally-equal SB3 defaults is detectable.
- [ ] Ensure MaskablePPO and plain PPO receive equivalent supported settings.
- [ ] Bump artifact/config schema versions because prior `config_used.json` files do not prove these coefficients were active.

## Validation plan

Instantiate PPO under a test config with deliberately distinctive values (`ent_coef=0.123`, `target_kl=0.00456`, `vf_coef=0.789`, etc.) and assert the underlying SB3 object exposes exactly those values. Separately, iterate every leaf key in shipped YAML files through a consumer/validation registry and fail the test for unregistered keys.

**Comment by ankaggarwal94** [comment]
## Independent CoVe revalidation

**Verdict:** CONFIRMED at `main` `5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`  
**Severity:** P1 retained  
**Confidence:** high

The shipped YAML declares `gae_lambda`, `clip_ratio`, `value_coef`, `entropy_coef`, `max_grad_norm`, and `target_kl`. `scripts/train_ppo.py` and `PPOBuzzer.__init__()` forward none of those six. The current Stable-Baselines3 PPO signature defaults to `gae_lambda=0.95`, `clip_range=0.2`, `ent_coef=0.0`, `vf_coef=0.5`, `max_grad_norm=0.5`, and `target_kl=None`: https://stable-baselines3.readthedocs.io/en/master/modules/ppo.html#stable_baselines3.ppo.PPO

That yields an important nuance:

- four ignored values happen to equal current SB3 defaults (`gae_lambda`, `clip_range`, `vf_coef`, `max_grad_norm`);
- two materially disagree with the repository config (`entropy_coef: 0.01` versus `ent_coef=0.0`, and `target_kl: 0.03` versus `None`).

The issue already states the material discrepancy correctly. Its stronger configuration-contract claim also survives: an override to any ignored field can be persisted while leaving the instantiated policy unchanged, and `merge_overrides()` creates unknown paths such as `environment.wait_penatly` rather than rejecting them.

**Disconfirmation/mitigation found:** some declared values currently work only by coincidence with third-party defaults. That reduces the number of presently divergent coefficients, but it does not make `config_used.json` evidence of the executed model, and it makes future library-default changes especially hazardous.

No duplicate issue was found. Issue #26 covers an invalid help example for override syntax, not unknown-key acceptance or dead runtime settings.

A fresh SB3 instantiation was not run in this connector-only audit; the repository call signature and official SB3 constructor defaults establish the discrepancy directly.
